blivet/size.py

```diff
@@ -19,229 +19,20 @@
-    if not six.PY2 and isinstance(s, bytes):
-        s = s.decode(sys.getdefaultencoding())
-    return s.translate(_ASCIIlower_table) # pylint: disable=no-member
-
-def _makeSpec(prefix, suffix, xlate, lowercase=True):
-    """ Synthesizes a whole word from prefix and suffix.
-
-        :param str prefix: a prefix
-        :param str suffixes: a suffix
-        :param bool xlate: if True, treat as locale specific
-        :param bool lowercase: if True, make all lowercase
-
-        :returns:  whole word
-        :rtype: str
-    """
-    if xlate:
-        word = (_(prefix) + _(suffix))
-        return word.lower() if lowercase else word
-    else:
-        word = prefix + suffix
-        return _lowerASCII(word) if lowercase else word
-
-def unitStr(unit, xlate=False):
-    """ Return a string representation of unit.
-
-        :param unit: a named unit, e.g., KiB
-        :param bool xlate: if True, translate to current locale
-        :rtype: some kind of string type
-        :returns: string representation of unit
-    """
-    return _makeSpec(unit.abbr, _BYTES_SYMBOL, xlate, lowercase=False)
-
-def parseUnits(spec, xlate):
-    """ Parse a unit specification and return corresponding factor.
-
-        :param spec: a units specifier
-        :type spec: any type of string like object
-        :param bool xlate: if True, assume locale specific
-
-        :returns: a named constant corresponding to spec, if found
-        :rtype: _Prefix or NoneType
-
-        Looks first for exact matches for a specifier, but, failing that,
-        searches for partial matches for abbreviations.
-
-        Normalizes units to lowercase, e.g., MiB and mib are treated the same.
-    """
-    if spec == "":
-        return B
-
-    if xlate:
-        spec = spec.lower()
-    else:
-        spec = _lowerASCII(spec)
-
-    # Search for complete matches
-    for unit in [_EMPTY_PREFIX] + _BINARY_PREFIXES + _DECIMAL_PREFIXES:
-        if spec == _makeSpec(unit.abbr, _BYTES_SYMBOL, xlate) or \
-           spec in (_makeSpec(unit.prefix, s, xlate) for s in _BYTES_WORDS):
-            return unit
-
-    # Search for unambiguous partial match among binary abbreviations
-    matches = [p for p in _BINARY_PREFIXES if _makeSpec(p.abbr, "", xlate).startswith(spec)]
-    if len(matches) == 1:
-        return matches[0]
-
-    return None
-
-def parseSpec(spec):
-    """ Parse string representation of size.
+from gi.repository import ByteSize
+from gi.repository.ByteSize import ROUND_DOWN, ROUND_UP
 
-        :param spec: the specification of a size with, optionally, units
-        :type spec: any type of string like object
-        :returns: numeric value of the specification in bytes
-        :rtype: Decimal
+# we just need to make these objects available here
+# pylint: disable=unused-import
+from gi.repository.ByteSize import B, KiB, MiB, GiB, TiB, PiB, EiB, ZiB, YiB, KB, MB, GB, TB, PB, EB, ZB, YB
```

Comment:
Not sure what you are asking about, but the point here is to get those "objects" (integer constants actually) to the namespace of this module because they were/are part of the API and other modules import and use them.